GitUI/Avatars/AvatarDownloader.cs

```diff
@@ -152,7 +152,11 @@ async Task<Image> Download()
                         {
                             webClient.Proxy.Credentials = CredentialCache.DefaultCredentials;
 
-                            using (var imageStream = await webClient.OpenReadTaskAsync(imageUrl))
+                            // actually, due to a .Net FX bug, WebClient.OpenReadTaskAsync *can* block before returning a Task
+                            // let's make sure this doesn't happen on the current thread (likely the UI thread)
+                            var readTask = Task.Run(() => webClient.OpenReadTaskAsync(imageUrl));
+
+                            using (var imageStream = await readTask)
                             {
                                 return Image.FromStream(imageStream);
```

Comment:
```c#
using (var imageStream =  await readTask)
		{
			using (var img = System.Drawing.Image.FromStream(imageStream)
			{
				return (Image)img.Clone();
			}
		}
```

https://docs.microsoft.com/en-us/dotnet/api/system.drawing.image.fromstream?view=netframework-4.7.2#System_Drawing_Image_FromStream_System_IO_Stream_  

Remarks
You must keep the stream open for the lifetime of the Image.

Reply:
It creates a GDI+ handle.  Since the docs say it and the code [here](https://referencesource.microsoft.com/#System.Drawing/commonui/System/Drawing/Image.cs,249).  Since we don't know how GDI+ works and how cleanup works in GDI+ and they did mention it in the docs, I don't see a problem of using clone to make sure.  The fact you were touching the code made me see the issue.  Yes we will need to visit all image.fromstream calls.

Reply:
Also [this] (https://referencesource.microsoft.com/#System.Drawing/commonui/System/Drawing/Image.cs,363) is where cleanup occurs.  So my using on the image does cleanup the image from the stream and we have an image that we don't have to worry about the stream closing mess. 

Dispose() passes true into Dispose(bool).

Reply:
This is why:

GDI+, and therefore the System.Drawing namespace, may defer the decoding of raw image bits until the bits are required by the image. Additionally, even after the image has been decoded, GDI+ may determine that it is more efficient to discard the memory for a large Bitmap and to re-decode later. Therefore, GDI+ must have access to the source bits for the image for the life of the Bitmap or the Image object.

Reply:
Doing more research.  Looks like clone might not be enough.  https://stackoverflow.com/questions/12709360/whats-the-difference-between-bitmap-clone-and-new-bitmapbitmap/13935966#13935966

Reply:
Anyway, assuming something has to be done (not 100% clear : this pattern exists as it is since at least 8d27233fbae40579f4299f95aa3be5251d14caf8, 8 years ago, apparently without actual issue), it should be in a separate PR, shouldn't it ?

Reply:
Alright.  We will get it from the [horse's mouth](https://github.com/dotnet/dotnet-api-docs/issues/1320).  Then we can raise an issue to fix it based on their feedback.

Reply:
> Anyway, assuming something has to be done (not 100% clear : this pattern exists as it is since at least [8d27233](https://github.com/gitextensions/gitextensions/commit/8d27233fbae40579f4299f95aa3be5251d14caf8), 8 years ago, apparently without actual issue), it should be in a separate PR, shouldn't it ?

Yes, I agree, this PR is about blocking UI.